### DISCUSSION TITLE
Is SQLAlchemy accessing the database securely?

### DISCUSSION THREAD

**Discussion by stevesmit** [sqlalchemy/sqlalchemy#6264] [Usage Questions] [answered]
**Describe your question**
In SQLAlchemy, when you create an engine:
```
import sqlalchemy as sa

params = <the parameters to connect to the DB such as the server and password, as a string>
engine = sa.create_engine("mssql+pyodbc:///?odbc_connect={}".format(params))
```

Is it, by default, accessing the database via a secure connection? For example, is the connection string being sent to the server in plaintext or anything like that? Is it encrypted? Also what protocol is being used...HTTPS? This may be a driver specific question. 

**Versions**
 - OS: Window 10
 - Python: 3.8.5
 - SQLAlchemy: 1.3.19
 - Database: Azure SQL Server 
 - DBAPI: ?

**Comment by zzzeek** [comment, marked as answer]
hi - 

you should consult the [pyodbc documentation](https://github.com/mkleehammer/pyodbc/wiki) for this information, SQLAlchemy merely passes through the parameters given.
